src/Symfony/Component/Form/Form.php

```diff
@@ -537,6 +537,9 @@ public function submit($submittedData, $clearMissing = true)
             $submittedData = null;
         } elseif (is_scalar($submittedData)) {
             $submittedData = (string) $submittedData;
+        } elseif (\is_array($submittedData) && !$this->config->getCompound() && !$this->config->hasOption('multiple')) {
```

Comment:
What about:
```php
elseif (\is_array($submittedData) && !$this->config->getCompound() && !$this->config->hasOption('multiple') &&  !$this->config->getOption('multiple'))
```
?

Reply:
It means we should add a test for such cases.

Reply:
Revert to my initial patch: `getOption('multiple')`

Reply:
Why do we need to check for the `multiple` option here? Such a hard coded knowledge about type specific options looks suspicious to me.